ironfish/src/storage/levelup/store.ts

```diff
@@ -104,16 +117,40 @@ export class LevelupStore<Schema extends DatabaseSchema> extends DatabaseStore<S
           continue
         }
 
-        yield [this.decodeKey(key), value as SchemaValue<Schema>]
+        if (keyRange?.ordered) {
+          cacheElements.push({ key: key, value: value as SchemaValue<Schema> })
+        } else {
+          yield [this.decodeKey(key), value as SchemaValue<Schema>]
+        }
       }
     }
 
+    let currElement = 0
+
     for await (const [key, value] of this.db.getAllIter(keyRange)) {
-      if (seen.has(key)) {
+      while (
+        currElement < cacheElements.length &&
+        (keyRange?.reverse
+          ? key.compare(cacheElements[currElement].key) < 0
+          : key.compare(cacheElements[currElement].key) > 0)
+      ) {
+        yield [this.decodeKey(cacheElements[currElement].key), cacheElements[currElement].value]
+        currElement++
+      }
+
+      if (currElement < cacheElements.length && key.equals(cacheElements[currElement].key)) {
+        yield [this.decodeKey(cacheElements[currElement].key), cacheElements[currElement].value]
+        currElement++
```

Comment:
Could we get rid of this block if we changed the compare functions above to use `<= 0` and `>= 0`?

Reply:
Pulled out this work into a different PR and made the change there - https://github.com/iron-fish/ironfish/pull/2799